### PR TITLE
Firefox scroll support for chat

### PR DESCRIPTION
**Pull request by litpub**
Attempts a fix for https://github.com/urbit/urbit/issues/2024 to support scrolling in chat for Firefox.

Currently, the flex-box layout uses `column-reverse` to implement the expected behaviors of a chat interface. The problem is that Firefox has a 5-year-old bug relating to the use of scroll support and `column-reverse`.

One approach, this one, is to just use `column` instead, and to deal with having to force scroll events on new messages.

This PR:

- Works by just using column
- un-reverses messages, now that `column-reverse` is gone
- Adds a named parameter to `scrollToBottom` so that you can control animation
- Calls `scrollToBottom` in `componentDidUpdate` so that we always scroll to the bottom on new messages
- Reorders the scroll element so that it appears below the message fragments in the DOM, as a consequence of dropping `column-reverse`
- Tidies up `render` a bit, extracts building the messages array into `messages`
- *Does not* implement the scroll lock behavior of chrome/safari on Firefox - I thought the existing function was too long, and should be refactored, and I wanted to get feedback first

Notable changes and tradeoffs:

1. New chats will render the first message at the top of screen, like so
Screenshot 2020-02-14 13 09 38

2. Others that I might not know about?

If this is a bad approach, lmk, I'm not fragile - I just want to fix the damn thing so I can use FF.

### CODE REVIEW

**Comment by litpub** [comment]
Also should I include the built pkg/arvo/app/chat/js/index.js or not? Seems kind of dangerous in a PR.

**Comment by loganallenc** [comment]
thanks! first off, run gulp bundle-prod and do a commit --amend and force push so that you're only committing the minified version of index.js

**Comment by litpub** [comment]
@loganallenc done

File: pkg/interface/chat/src/js/components/chat.js [hunk 1]
```diff
@@ -168,13 +175,10 @@ export class ChatScreen extends Component {
    }
  }
 
- render() {
+ messages() {
```
Comment (loganallenc) [inline pkg/interface/chat/src/js/components/chat.js]: Naming. messagesWithPending() would be better

**Comment by matildepark** [comment]
> New chats will render the first message at the top of screen, like so

Have you tried `align-items: flex-end` (should just be `items-end` in indigo-static classes)? Unsure if that would introduce issues with a full chatlog but might work.

**Comment by litpub** [comment]
Thanks for round of feedback all - will push some changes shortly.

**Comment by benjamin-tlon** [comment]
Hell yes.

**Comment by loganallenc** [comment]
@litpub any word? no rush, just don't want to drop this.

**Comment by litpub** [comment]
@loganallenc revisiting now

**Comment by litpub** [comment]
@loganallenc force pushed - also the CSS file is diffing now, but I assume that's because a PR was merged without gulp build production somewhere else.

**Pull request merged**